File: cpp/questions/findKthLargest.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        std::priority_queue<int, std::vector<int>, std::greater<>> pq;        
        for (const auto n: nums) {
            if (pq.size() < k) {
                pq.push(n);
            } else if (n > pq.top()) {
                pq.pop();
                pq.push(n);
            }
        }
        return pq.top();
    }
};

File: python/questions/findKthLargest.py
from typing import List
import heapq

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        hq = []
        for n in nums:
            if len(hq) < k:
                heapq.heappush(hq, n)
            elif n > hq[0]:
                heapq.heappop(hq)
                heapq.heappush(hq, n)
        return hq[0]
